pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -127,6 +128,7 @@
 
 assert stdenv.cc.libc or null != null;
 assert pipewireSupport -> !waylandSupport || !webrtcSupport -> throw "pipewireSupport requires both wayland and webrtc support.";
+assert pgoSupport -> !ltoSupport -> throw "PGO requires LTO, and currently only works on x86_64-linux.";
```

Comment:
Taken from https://github.com/NixOS/nixpkgs/pull/129503/files#diff-e3657e151bc76a990d176c9eb753b319e2e39ca4bf1732a5c705ed060ea3a6b0R78

I'm not sure whether mentioning the platform here is really necessary, nor why LTO would be a requirement.

Reply:
Might be a bit late but perhaps you could add a `firefox` prefix to these throws?

There is nothing worse than a random throw without any indication what package it might be from. Having to grep through nixpkgs shouldn't be the default.

Reply:
That check was due to the profdata being sourced from upstream. LTO changes stdenv to provide clang, which in turn allowed for the profdata to be reused from upstream. Since profdata is being generate from source the check does not matter here.

The configure error is because stdenv falls back to GCC when LTO is disabled. Ideally clang would be the default for LTO and non-LTO builds.